Functionality branch

CharlesNunn001 opened the pull request:
Sorted a few things out - shown in commit messages

CharlesNunn001 committed:
Added slightly modified NcbiEutils, loader and model modules, added main() script framework and error function, both WIP

CharlesNunn001 committed:
Restructured to contain argparse script, main module 'validation' and removing unnecessary/unused files

CharlesNunn001 committed:
Changed errors to human readable, changed list stats to be part of spreadsheet class and all calls to this, added additional presence check for 'SUPPLIER SAMPLE NAME' to allow a row to be taken in

puethe commented:
Hi Charles,
this looks good, I like the new class and the new error messages. Some comments below.
- remove .idea (and add to .gitignore to avoid adding it accidentally in the future)
- remove fake_manifest.xlsx, or move to test file directory if you're using it in tests
- scripts/manifest-validator.py: I would suggest to remove the ".py" in the filename, because this is a script you're calling, and the shebang line determines how it is executed
- don't call your main module "modules". That might cause conflicts. Use a specific name.
- similarly, don't call your main function "main", but use a descriptive name
- class Spreadsheet: I'd suggest to define an __init__ function instead of new_instance. Also, what does this class provide? There's no documentation.
- class Spreadsheet: I'd also suggest to save the sample ID instead of the row number as unique identifier. Easier to trace for the user.
- class SpreadsheetLoader: don't use "i" as variable name, not even for counters. You have tab completion, so there's no need in using extra-short variable names. What does i stand for? if it's a row index, call it row_index. Etc.
- class SpreadsheetLoader: There's some code duplication in lines 34-39 that I'd suggest to avoid. save "self._sheet.cell_value(header_row, i)" into a temporary variable that has an informative name. Also, use if..else if where appropriate to clarify that the options are mutually exclusive
- class SpreadsheetLoader, function __extract_float_value: the name of the function does not fit what it's doing. It in fact extracts an integer converted to a string. Why not to just save the taxon ID as an integer (and rename the function accordingly)? You can fix the data type of your function input/output (and in fact of every variable), and I'll show you how to do that. Adds clarity and avoids weird execution.
- validation.py, function report_error: this should just return a string, which is then added to the list by the calling function (if you want to do this at all, and not just print it out directly)
- main.py, line 12: I don't think this statement makes much sense. Probably a leftover from an older setup.
- tests are failing - see the external-import for travis setup

CharlesNunn001 committed:
'.py' removed from script name and .idea removed from repo
'modules' --> validation_components and main --> validation_runner
Spreadsheet --> ManifestEntry now uses __init__ and sample_id replaced row
SpreadsheetLoader counters made meaningful
SpreadsheetLoader __extract_cell_value: _text_value and _float_value merged together
validation report_error returns string
validation_runner line 12 switched to class format

CharlesNunn001 committed:
Added codecov and travis requirements
